### PR TITLE
Feature/lost package management

### PR DESCRIPTION
**Pull request by supertoto29**
**Warning :** Before creating a pull request, make sure that it does not already exists in the [pull requests](../). Thank you.

## Description
Added the package lost recovery on the master sensor. The slave doesn't need this recovery. It will resend the message.
Added a mutex to protect the writerQueue for the ROS Service to function properly.

## How has this been tested ?
Tested locally. Couldn't test the communication between 2 sensors.

## Checklist
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

### CODE REVIEW

File: src/sharedQueue.h [hunk 1]
```diff
@@ -0,0 +1,142 @@
+/**
+ * \file	serial.h
+ * \author	Lucas Mongrain
```
Comment (0x72D0) [inline src/sharedQueue.h:3]: I appreciate, but I didn't make this one, only modify it a little: https://stackoverflow.com/questions/36762248/why-is-stdqueue-not-thread-safe

**Review by 0x72D0 — CHANGES_REQUESTED** [review]

File: src/sharedQueue.h (hunk 1 above)
Comment (supertoto29) [reply src/sharedQueue.h]: Change it with a direct link to the forum

File: src/sharedQueue.h [hunk 2]
```diff
@@ -0,0 +1,142 @@
+/**
+ * \file	serial.h
+ * \author	user66875 (https://stackoverflow.com/a/36763257)
```
Comment (Kam0ss) [inline src/sharedQueue.h:3]: Are we sure we want to put it like this ? 
And is the whole header license really useful ?

**Review by Kam0ss — APPROVED** [review]
LGTM

File: src/sharedQueue.h (hunk 2 above)
Comment (supertoto29) [reply src/sharedQueue.h:3]: Since I used some codes that were already GPL, we better add it

**Review by 0x72D0 — APPROVED** [review]

**Pull request merged**